src/inputs/ChipSelectField.tsx

```diff
@@ -0,0 +1,194 @@
+    onFocus,
+    onBlur,
+    clearable = false,
+  } = props;
+  const tid = useTestIds(props, "chipSelectField");
+  const typeScale = fieldProps?.typeScale ?? "sm";
+  const showClearButton = clearable && !!value;
+
+  const [isFocused, setIsFocused] = useState(false);
+  const [isClearFocused, setIsClearFocused] = useState(false);
+  const { focusProps } = useFocus({
+    onFocus: (e) => {
+      // Do not call `onFocus` if focus was returned to the trigger element from the Popover
+      if (popoverRef.current && popoverRef.current.contains(e.relatedTarget as HTMLElement)) {
+        return;
+      }
+
+      maybeCall(onFocus);
+    },
+    // Do not call onBlur if we just opened the menu
+    onBlur: () => !state.isOpen && maybeCall(onBlur),
+    // Do not change focus state if select menu is opened
+    onFocusChange: (isFocused) => !state.isOpen && setIsFocused(isFocused),
+  });
+  const { focusProps: clearFocusProps } = useFocus({ onFocusChange: setIsClearFocused });
+
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const popoverRef = useRef<HTMLDivElement>(null);
+  const listBoxRef = useRef(null);
+
+  const chipStyles = Css[typeScale].bgGray300.gray900.br16.pxPx(10).pyPx(2).$;
+
+  const selectChildren = useMemo(
+    () =>
+      options.map((o) => (
+        <Item key={valueToKey(getOptionValue(o))} textValue={getOptionLabel(o)}>
+          <span css={chipStyles}>{getOptionLabel(o)}</span>
+        </Item>
+      )),
+    [options],
+  );
+
+  const state = useSelectState<any>({
+    label,
+    children: selectChildren,
+    autoFocus: false,
+    selectedKey: valueToKey(value),
+    onSelectionChange: (key) => {
+      const selectedItem = options.find((o) => getOptionValue(o) === key);
+      if (selectedItem) {
+        onSelect(key as V, selectedItem);
+      }
+    },
+    onOpenChange: (isOpen) => {
+      if (!isOpen && buttonRef.current) {
+        // When closing reset the focus to the button element.
+        buttonRef.current.focus();
+      }
+    },
+  });
+
+  const { labelProps, triggerProps, valueProps, menuProps } = useSelect(
+    { label, children: selectChildren },
+    state,
+    buttonRef,
+  );
+
+  const { buttonProps } = useButton(triggerProps, buttonRef);
+  const { overlayProps: positionProps } = useOverlayPosition({
+    targetRef: buttonRef,
+    overlayRef: popoverRef,
+    shouldFlip: true,
+    scrollRef: listBoxRef,
+    isOpen: state.isOpen,
+    onClose: state.close,
+    placement: "bottom left",
+  });
+
+  positionProps.style = {
+    ...positionProps.style,
+    width: buttonRef?.current?.clientWidth,
+    // Ensures the menu never gets too small.
+    minWidth: 200,
+  };
+
+  return (
+    <>
+      <div
+        css={{
+          ...chipStyles,
+          ...Css.dif.p0.mwPx(32).if(!value).bgGray200.$,
+          ...(isFocused ? Css.bshFocus.$ : {}),
+        }}
+      >
+        <Label label={label} labelProps={labelProps} hidden {...tid.label} />
+        <button
```

Comment:
Is this the rounded button?

Reply:
Yes. This button matches the designs for the Chips with just a different color to follow the SchedulesV2 design. This is a prototype component, so not forcing it to comply with existing styles (but it does for the most part).